src/main/java/org/elasticsearch/index/engine/ShadowEngine.java

```diff
@@ -155,7 +158,9 @@ public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineExcepti
         } finally {
             store.decRef();
         }
-        return id;
+        // We can just return an empty commit ID since this is a read only engine that
+        // doesn't modify anything so the content of this ID doesn't really matter
```

Comment:
can we add a little note about avoiding race conditions between read and writes on a shared FS ? I think this will people we not see why we opted to EMPTY_COMMIT_ID